scripts/ci/source_code_static_analysis.py

```diff
@@ -0,0 +1,154 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+
+import os
+import sys
+import multiprocessing
+
+from pylint import lint
+from flake8.main import application
+
+from util import get_repo_root
+
+# import argparse
+# from verify_codeowners import main as _verify_codeowners
+# from verify_license import main as _verify_license
+
+
+SDK_FILE_MARKER = r"# Code generated by Microsoft (R) AutoRest Code Generator."
+
+
+# build list of sdk files
+def _get_sdk_module_list(root_dir):
+    dir_paths = [os.path.join(root_dir, path) for path in os.listdir(root_dir)]
+    sdk_files = []
+
+    for path in dir_paths:
+        if os.path.isdir(path):
+            sdk_files.extend(_get_sdk_module_list(path))
+        elif _is_sdk_file(path):
+            sdk_files.append(os.path.dirname(path))
+
+    return sdk_files
+
+
+# check if the current file is a python sdk file
+def _is_sdk_file(file_path):
+    # don't bother opening non-python files. e.g pyc files.
+    if not file_path.endswith(".py"):
+        return False
+    with open(file_path, encoding='utf-8') as f:
+        for line in f:
+            if SDK_FILE_MARKER in line:
```

Comment:
Just one thing. Since we're trying to codegen tool to generate azure cli in the near future. Will this `SDK_FILE_MARKER` appears too?

Reply:
Since you're trying to ignore these files. Right?

Reply:
Looks so.

Reply:
This is copy from scripts/ci/test_static.py